runners/direct-java/src/main/java/org/apache/beam/runners/direct/StepTransformResult.java

```diff
@@ -127,5 +148,10 @@ public Builder addOutput(Collection<UncommittedBundle<?>> outputBundles) {
       bundlesBuilder.addAll(outputBundles);
       return this;
     }
+
+    public Builder withAdditionalOutput(boolean producedAdditionalOutput) {
```

Comment:
Also I think this is essentially only used with `true` except in one programmatic case. Note that the enum might form a lattice where you add types of output to it, so you can ask "did it produce elements?" "did it write somewhere as a side effect"? etc